kore/src/Kore/Internal/Conditional.hs

```diff
@@ -277,28 +282,48 @@ instance
         unparseConditional
             sort
             (unparse term)
-            (unparse termLikePredicate)
+            (unparse predicate)
```

Comment:
That's fair enough, but it turns out that this instance and the one in `Predicate.hs` are used in lots of places (any place where we want to unparse something containing a Predicate, like an Equation), mainly for logging and error printing. I tried removing these instances and then I ended up having to remove a lot more, so I'm not sure if it would be wise to go down that path.